### PR TITLE
feat: add support for RollupCreator v2.1

### PR DESCRIPTION
**Pull request by spsjvc**
Closes FS-654.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -5,7 +5,7 @@ BASESCAN_API_KEY=
 NITRO_TESTNODE_DEPLOYER_PRIVATE_KEY=0xb6b15c8cb491557369f3c7d2c287b053eb229daa9c22138887752191c9520659
 
 NITRO_TESTNODE_L2_ROLLUP_OWNER_PRIVATE_KEY=0xdc04c5399f82306ec4b4d654a342f40e2e0620fe39950d967e1e574b32d4dd36
-NITRO_TESTNODE_L2_TOKEN_BRIDGE_DEPLOYER_PRIVATE_KEY=0xadd3d9301e184194943ce6244aa25c90e73c5843db16a994d202091f97f5bb27
+NITRO_TESTNODE_L2_TOKEN_BRIDGE_DEPLOYER_PRIVATE_KEY=0xb6b15c8cb491557369f3c7d2c287b053eb229daa9c22138887752191c9520659
```
Comment (spsjvc) [inline .env.example:8]: changed based on input from Goran

File: src/createRollup.integration.test.ts [hunk 2]
```diff
@@ -54,25 +58,24 @@ describe(`create an AnyTrust chain that uses ETH as gas token`, async () => {
 });
 
 describe(`create an AnyTrust chain that uses a custom gas token`, async () => {
-  // deployed during nitro testnode running process
-  const customGasTokenAddress = '0xc57a290f65F1D433f081381B2A7A523Ea70f1134';
+  const nativeToken = getInformationFromTestnode().l3NativeToken;
```
Comment (spsjvc) [inline src/createRollup.integration.test.ts:61]: read value dynamically instead of hardcoding

File: wagmi.config.ts [hunk 3]
```diff
@@ -156,8 +177,8 @@ const contracts: ContractConfig[] = [
       [arbitrumSepolia.id]: '0x56C486D3786fA26cc61473C499A36Eb9CC1FbD8E',
       [baseSepolia.id]: '0xFC71d21a4FE10Cc0d34745ba9c713836f82f8DE3',
       // local nitro-testnode (on "release" branch with --tokenbridge --l3node --l3-token-bridge flags)
-      [nitroTestnodeL1.id]: '0x54B4D4e578E10178a6cA602bdb6df0F213296Af4',
-      [nitroTestnodeL2.id]: '0x38f35af53bf913c439eab06a367e09d6eb253492',
+      [nitroTestnodeL1.id]: '0x4Af567288e68caD4aA93A272fe6139Ca53859C70',
+      [nitroTestnodeL2.id]: '0x38F35Af53bF913c439eaB06A367e09D6eb253492',
```
Comment (spsjvc) [inline wagmi.config.ts:181]: adresses updated in the latest testnode

File: src/isAnyTrust.ts [hunk 4]
```diff
@@ -18,12 +26,24 @@ export async function isAnyTrust<TChain extends Chain | undefined>({
...
+  } catch (error) {
+    // try parsing for RollupCreator v1.1
+    const {
+      args: [{ config }],
+    } = decodeFunctionData({
+      abi: [createRollupV1Dot1ABI],
+      data: transaction.input,
+    });
+    return parseConfig(config);
+  }
```
Comment (fionnachan) [inline src/isAnyTrust.ts:48]: this works but looks pretty ugly

alternatively we can loop through an array of ABI to try parsing?

File: src/isAnyTrust.unit.test.ts [hunk 5]
```diff
@@ -16,7 +17,20 @@ it('should return true for AnyTrust chain', async () => {
   expect(isPlaynanceAnyTrust).toBeTruthy();
 });
 
-it('should return false for non AnyTrust chain', async () => {
+it('should return true for AnyTrust chain (RollupCreator v2.1)', async () => {
+  const client = createPublicClient({
+    chain: arbitrumSepolia,
+    transport: http(),
+  });
+  // https://sepolia.arbiscan.io/tx/0xc1d9513cee57252ab9a0987e3ac4bf23aca7f5c58478a29439ecb1ef815cd379
```
Comment (fionnachan) [inline src/isAnyTrust.unit.test.ts:25]: this chain's deploy param has lots of batch posters which is great for testing

i don't see this tx used for any other tests in the repo for the new batch posters [], perhaps you can use it on other tests?

File: wagmi.config.ts [hunk 6]
```diff
@@ -139,6 +139,27 @@ const contracts: ContractConfig[] = [
...
+      [sepolia.id]: '0xfb774ea8a92ae528a596c8d90cbcf1bdbc4cee79',
+      [holesky.id]: '0x03c70F125Df471E4fd0515ca38504edFE6900F19',
+      // testnet L2
+      [arbitrumSepolia.id]: '0xd2Ec8376B1dF436fAb18120E416d3F2BeC61275b',
+      [baseSepolia.id]: '0x6e244cD02BBB8a6dbd7F626f05B2ef82151Ab502',
+      // local nitro-testnode (on "release" branch with --tokenbridge --l3node --l3-token-bridge flags)
+      [nitroTestnodeL1.id]: '0x82a3c114b40ecf1fc34745400a1b9b9115c33d31',
+      [nitroTestnodeL2.id]: '0x4287839696d650a0cf93b98351e85199102335d0',
+    },
+  },
```
Comment (fionnachan) [inline wagmi.config.ts:162]: this is also hardcoded as `rollupCreatorAddress` in `RollupCreator/v2.1.ts`, possible to consolidate?

File: .github/workflows/build-test.yml [hunk 7]
```diff
@@ -69,16 +69,10 @@ jobs:
...
         with:
-          nitro-testnode-ref: 1647de2c5f75ee0fbe5986e3f20d35bc75191ea6 # https://github.com/OffchainLabs/nitro-testnode/pull/42
-          # Use simple mode
-          no-simple: false
-          # RollupCreator on L1 is deployed by default
-          # RollupCreator on L2 is deployed with --l3node
-          l3-node: true
-          # L3 node with custom fee token when using --l3-fee-token
-          args: --l3-fee-token
+          nitro-testnode-ref: executor-owner
```
[the remark this answers is not in the document]
Comment (spsjvc) [reply .github/workflows/build-test.yml:74]: yup

File: src/isAnyTrust.unit.test.ts (hunk 5 above)
Comment (spsjvc) [reply src/isAnyTrust.unit.test.ts:25]: added to Linear

**Review by fionnachan — APPROVED** [review]

**Pull request merged**